### ISSUE TITLE
how to fine tune with the gpt2?what's the dataset format with my own data

### ISSUE THREAD

**Issue by ucas010** [huggingface/transformers#21591]
### System Info

transformers-cli env

Copy-and-paste the text below in your GitHub issue and FILL OUT the two last points.

- `transformers` version: 4.26.0.dev0
- Platform: Linux-3.10.0-1160.81.1.el7.x86_64-x86_64-with-glibc2.17
- Python version: 3.9.15
- Huggingface_hub version: 0.11.1
- PyTorch version (GPU?): 1.13.1+cu116 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

### Who can help?

@sgugger @Rocketknight1 
@gmftbyGMFTBY 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

https://github.com/huggingface/transformers/tree/main/examples/pytorch/text-generation
python run_generation.py \
    --model_type=gpt2 \
    --model_name_or_path=gpt2
    ???
   --train_file ?
   --valid_file?

### Expected behavior

fine tuned gpt2 will generate chinese well

**Comment by ydshieh** [comment]
@ucas010  The script you linked is not for training/fine-tuing however. It's only for generation with a provided prompt.

**Comment by ucas010** [comment]
@ydshieh year，I have tried with Chinese，but the result is not good , you see 
02/14/2023 09:52:08 - INFO - __main__ - Namespace(model_type='gpt2', model_name_or_path='gpt2', prompt='', length=20, stop_token=None, temperature=1.0, repetition_penalty=1.0, k=0, p=0.9, prefix='', padding_text='', xlm_language='', seed=42, no_cuda=False, num_return_sequences=1, fp16=False, device=device(type='cuda'), n_gpu=4)
Model prompt >>> 刑事责任，民事责任，刑事诉讼案件，简易程序，人民法院
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.
=== GENERATED SEQUENCE 1 ===
刑事责任，民事责任，刑事诉讼案件，简易程序，人民法院，傳，文愛民要也�

**Comment by ydshieh** [comment]
@ucas010 , as sgugger mentioned, this kind of question is for [Hugging Face Forums](https://discuss.huggingface.co/).
The GitHub repository  is only for issues, bugs or features in the library.

I am going to close this issue. But FYI: GPT-2 is only trained on English corpus, and you can't use it for other languages (not even with just fine-tuning).
